### PR TITLE
Implement THIS_MODULE.

### LINKED ISSUE #10: Implement `THIS_MODULE`
...so that the kernel module developer can pass it to the kernel APIs that use it.

### PR DESCRIPTION
**Pull request by alex**
Closes #15
Fixes #10

### CODE REVIEW

**Comment by alex** [comment]
```
error[E0425]: cannot find value `__this_module` in module `kernel::bindings`
```

Hmm, anyone have a clue what happened that would lead to this?

**Comment by ojeda** [comment]
The `-DMODULE` flag is not there in `c_flags`, so `__this_module` is never seen by `bindgen`. If it was previously there at one point, it got probably lost in one of the refactors I did (or perhaps I took it out until it was actually needed and/or we discussed whether it was worth to have 2 sets of bindings, one for `MODULE` and one for `!MODULE`).

In `rust/Makefile`, please add it on line:

    --size_t-is-usize -o $@ -- $(bindgen_c_flags) -DMODULE

File: rust/kernel/lib.rs [hunk 1]
```diff
@@ -44,6 +44,18 @@ pub trait KernelModule: Sized + Sync {
     fn init() -> KernelResult<Self>;
 }
 
+/// An instance equivalent to `THIS_MODULE` in C code.
+pub struct ThisModule(*mut bindings::module);
+
+// Safe because THIS_MODULE may be used from all threads within a module.
```
Comment (ojeda) [inline rust/kernel/lib.rs]: Just a nit on the style to be consistent.
```suggestion
// SAFETY: `THIS_MODULE` may be used from all threads within a module.
```

**Comment by alex** [comment]
Nice catch, would have taken me a while to figure out.

**Comment by ojeda** [comment]
It failed now but not due to the `__this_module`, so that should be fine.

**Comment by alex** [comment]
Making progress! (My local dev env is messed up ATM, so I'm kind of relying on CI :-))

**Comment by ojeda** [comment]
I can see that :D

I'm sure GitHub doesn't mind being used as an IDE with all the "edit on the web" features they add here and there...

**Review by ojeda — APPROVED** [review]
Thanks for taking care of this one!

**Pull request merged**